TeamSnapSDK/Profiler/TSDKProfileTimer.m

```diff
@@ -37,35 +38,43 @@ -(instancetype) init {
     if (self) {
         _timers = [[NSMutableDictionary alloc] init];
         _totalTimes = [[NSMutableDictionary alloc] init];
+        _accessQueue = dispatch_queue_create("com.teamsnap.perf.timer", DISPATCH_QUEUE_SERIAL);
     }
     return self;
 }
 
 -(void) startTimeWithId:(id)timerId {
     if (timerId) {
-        [self.timers setObject:[NSDate date] forKey:timerId];
+        NSDate *timerDate = [NSDate date];
+        dispatch_sync(self.accessQueue, ^{
```

Comment:
Consumers of this API expect this function to block until completion. I went back and forth on async for this method. Of any of them, this could be async since this function doesn't actually return anything.